packages/block-library/src/block/edit.js

```diff
@@ -71,7 +87,49 @@ export default function ReusableBlockEdit( {
 	name,
 	attributes: { ref },
 	__unstableParentLayout: parentLayout,
+	context: { postId },
+	clientId: patternClientId,
 } ) {
+	const { setBlockEditingMode } = useDispatch( editorStore );
+	const { editUrl, innerBlocks } = useSelect(
+		( select ) => {
+			const { canUser } = select( coreStore );
+			const { getSettings, getBlocks } = select( editorStore );
+			const blocks = getBlocks( patternClientId );
+			const isBlockTheme = getSettings().__unstableIsBlockBasedTheme;
+			const defaultUrl = addQueryArgs( 'post.php', {
+				action: 'edit',
+				post: ref,
+			} );
+			const siteEditorUrl = addQueryArgs( 'site-editor.php', {
+				postType: 'wp_block',
+				postId: ref,
+				categoryType: 'pattern',
+				canvas: 'edit',
+				refererId: postId,
+			} );
+
+			// For editing link to the site editor if the theme and user permissions support it.
+			return {
+				innerBlocks: blocks,
+				editUrl:
+					canUser( 'read', 'templates' ) && isBlockTheme
+						? siteEditorUrl
+						: defaultUrl,
+			};
+		},
+		[ patternClientId, postId, ref ]
+	);
+
+	useEffect( () => {
+		innerBlocks.forEach( ( block ) => {
+			const editMode = isPartiallySynced( block )
+				? 'contentOnly'
+				: 'disabled';
+			setBlockEditingMode( block.clientId, editMode );
+		} );
```

Comment:
May need to do this recursively for 'contentOnly' as a content block could be nested in a group 😓

Reply:
ah, true, done.